penn/laundry.py

```diff
@@ -73,25 +80,32 @@ def hall_status(self, hall_no):
             raise ValueError("Room Number must be integer")
 
         r = requests.get(HALL_BASE_URL + str(num))
-        if r.status_code == 200:
-            parsed = BeautifulSoup(r.text, 'html5lib')
-            tables = parsed.find_all('table')
-            hall_name = tables[2].get_text().strip()
-            info_table = tables[4]
-            data = [list(filter(lambda x: len(x) > 0, [val.get_text().strip() for val in row.find_all('td')])) for row in info_table.find_all('tr')]
-            data_improved = list(filter(lambda x: len(x) > 0, data))[1:]
-
-            def toDict(data_row):
-                d = dict()
-                d['number'] = data_row[0]
-                d['machine_type'] = data_row[1]
-                d['available'] = data_row[2] == u'Available'
-                if len(data_row) == 4:
-                    d['time_left'] = data_row[3]
-                else:
-                    d['time_left'] = None
-                return d
-
-            return {'machines': list(map(toDict, data_improved)), 'hall_name': hall_name}
-        else:
-            return {'error': 'The laundry api is currently unavailable.'}
+        r.raise_for_status()
```

Comment:
Documentation is going to be needed for status errors at some point. I'll trello it.